packages/react/src/routing/NavManager.tsx

```diff
@@ -67,14 +67,22 @@ export class NavManager extends React.PureComponent<NavManagerProps, NavContextS
       resetTab: this.props.onResetTab,
     };
 
-    if (typeof document !== 'undefined') {
-      document.addEventListener('ionBackButton', (e: any) => {
-        e.detail.register(0, (processNextHandler: () => void) => {
-          this.nativeGoBack();
-          processNextHandler();
-        });
-      });
-    }
+    this.handleIonBackButton = this.handleIonBackButton.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('ionBackButton', this.handleIonBackButton);
```

Comment:
Two things here:

1. I would prefer if we kept this `addEventListener` in the constructor to avoid a race condition between when the instance is created and when it is mounted in the DOM (even as unlikely as it may be).
2. We need to keep the `typeof document !== 'undefined'` checks for both `addEventListener` and `removeEventListener` as both `document` and `window` will be undefined when doing SSR. Without this check an error will be thrown and SSR will fail.